meeteval/wer/matching/cy_levenshtein.pyx

```diff
@@ -113,59 +111,9 @@ def levenshtein_distance(
         reference, hypothesis, cost_del, cost_ins, cost_sub, cost_cor
     )
 
-cdef _validate_inputs(reference, hypothesis, vector[pair[double, double]] reference_timing, vector[pair[double, double]] hypothesis_timing):
-    if len(reference) != len(reference_timing):
-        raise ValueError(
-            f'reference and reference_timing have mismatching lengths '
-            f'{len(reference)} != {len(reference_timing)}'
-        )
-    if len(hypothesis) != len(hypothesis_timing):
-        raise ValueError(
-            f'hypothesis and hypothesis_timing have mismatching lengths '
-            f'{len(hypothesis)} != {len(hypothesis_timing)}'
-        )
-
-    # reference_timing = np.array(reference_timing)
-    # hypothesis_timing = np.array(hypothesis_timing)
-
-    # assert len(reference) == 0 or reference_timing.shape == (len(reference), 2), (
-    # reference_timing.shape, len(reference))
-    # assert len(hypothesis) == 0 or hypothesis_timing.shape == (len(hypothesis), 2), (
-    # hypothesis_timing.shape, len(hypothesis))
-    # assert len(reference) == 0 or len(hypothesis) == 0 or reference_timing.dtype == hypothesis_timing.dtype, (
-    # reference_timing.dtype, hypothesis_timing.dtype)
-
-    cdef double last_start = 0
-    for t in reference_timing:
-        if t.second < t.first:
-            raise ValueError(
-                f'The end time of an interval {t.second} must not be smaller than its begin time {t.first}, '
-                f'but the reference violates this'
-            )
-        if t.first < last_start:
-            raise ValueError(
-                f'The start times of the annotations must be increasing, which they are not for the reference. '
-                f'(found at least one interval where {t.first} < {last_start}) '
-                f'This might be caused by overlapping segments, see the (potential) previous warning.'
-            )
-        last_start = t.first
-
-    last_start = 0
-    for t in hypothesis_timing:
-        if t.second < t.first:
-            raise ValueError(
-                f'The end time of an interval {t.second} must not be smaller than its begin time {t.first}, '
-                f'but the hypothesis violates this'
-            )
-        if t.first < last_start:
-            raise ValueError(
-                f'The start times of the annotations must be increasing, which they are not for the hypothesis. '
-                f'(found at least one interval where {t.first} < {last_start}) '
-                f'This might be caused by overlapping segments, see the (potential) previous warning.'
-            )
-        last_start = t.first
 
-    return reference, hypothesis, reference_timing, hypothesis_timing
+cdef _timing_is_integer(timing):
+    return all(isinstance(t[0], int) and isinstance(t[1], int) for t in timing)
```

Comment:
What is the usual result? If it is True, a list comprehension is faster, otherwise the generator comprehension is faster. Or is it different in Cython?

Reply:
It's usually `False` when processing STMs since they usually use (floating-point) seconds as a unit. I don't know how this is translated to C++ by Cython, but I believe this still uses the Python functions.

Reply:
Ok, then the generator comprehension is fine.